lib/logstash/inputs/http.rb

```diff
@@ -75,11 +83,20 @@ def register
   end # def register
 
   def run(queue)
```

Comment:
regarding the rack vs custom code, I just made the decision since rack forces a lot of decisions on you and seemed overkill for this, but I can take a second look.